examples/configfile/custom/v1alpha1/types.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cfg "sigs.k8s.io/controller-runtime/pkg/config/v1alpha1"
+)
+
+// +kubebuilder:object:root=true
+
+// CustomControllerConfiguration is the Schema for the CustomControllerConfigurations API
+type CustomControllerConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// ControllerConfiguration returns the contfigurations for controllers
+	cfg.ControllerConfiguration `json:",inline"`
+
+	ClusterName string `json:"string,omitempty"`
+}
+
+// GetControllerConfiguration returns the configuration for controller-runtime
+func (c *CustomControllerConfiguration) GetControllerConfiguration() cfg.ControllerConfiguration {
```

Comment:
`cfg.ControllerConfiguration` implements this, so it shouldn't be necessary, right?

Reply:
Could also just run `c` if you wanted to override a function since this is an interface right?